lib/AST/FeatureSet.cpp

```diff
@@ -506,6 +506,26 @@ UNINTERESTING_FEATURE(Embedded)
 UNINTERESTING_FEATURE(Volatile)
 UNINTERESTING_FEATURE(SuppressedAssociatedTypes)
 
+static bool disallowFeatureSuppression(StringRef featureName, Decl *decl);
+
+static bool allBoundTypesAreCopyable(Type type, DeclContext *context) {
+  assert(type->getAnyNominal());
```

Comment:
You can replace this entire function with `lookupConformance(type, Copyable)`

Reply:
How so? I don't need to know whether the type `Optional<X>` within a function declaration is Copyable, or even conditionally copyable.

I'm looking to see if the type argument `X` is Copyable, because I know Optional will get printed twice:
```
#if ...
  enum Optional<Wrapped: ~Copyable> ...
#else
  enum Optional<Wrapped> ...
#endif
```

and I want to make sure that the function definition will work for both, which is when X is Copyable.

Reply:
So when you visit Optional<X>, you see that Optional is Copyable. Then you visit X and check if it’s Copyable; if not, you need a feature guard